astropy/io/registry/core.py

```diff
@@ -179,7 +184,8 @@ def read(self, cls, *args, format=None, cache=False, **kwargs):
         ctx = None
         try:
             # Expand a tilde-prefixed path if present in args[0]
-            args = _expand_user_in_args(args)
+            if len(args):
+                args = (_expand_user_in_arg0(args[0]), *args[1:])
```

Comment:
is args[0] always a str | os.PathLike[str] ?

Reply:
I think this and the previous implementation work even when args is not a str, so the annotations aren't quite correct.

Reply:
I'm not sure I understand what you mean. If `args[0]` is anything other than `os.PathLike[str]` (in practice, this means `pathlib.Path`), the annotation correctly reflects that the return type is the input type, doesn't it ?

Reply:
oops, I forgot about that. Then I guess the correct annotation is just an unbound `TypeVar` ?

Reply:
Except that Pathlike needs to be restricted. Might need to use an overload.